debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelope.java

```diff
@@ -50,7 +50,12 @@ public R apply(R r) {
             // update
             if (patchRecord.value() != null) {
                 valueDocument = BsonDocument.parse(patchRecord.value().toString());
-                valueDocument = valueDocument.getDocument("$set");
+                if(valueDocument.containsKey("string") && !valueDocument.containsKey("$set")){
```

Comment:
What is "string", the type of that field or its name? Either way it seems this fix would have to be applied in a more generic fashion than just for strings?

Reply:
You are right. In my test case, I only inserted one String field in mongo db which trigger the above oplog. Please refer to the following command. 

 db.kconnect_db.insert({"name" :"test"})

Reply:
Hum, ok. Seems that fix will have to be expanded than to cover all possible field types.

Reply:
Hey @axieyangb, how should we proceed with this one? Do you plan to provide a more general solution?

Reply:
I will do more experiments before fixing the issue. I am too busy recently, once I got time, I will provide a general solution. Thanks

Reply:
Hey @axieyangb, just checking in. Are you still interested in completing this work? Thanks!